drivers/clock_control/clock_control_rts5912_sccon.c

```diff
@@ -0,0 +1,242 @@
+		src = ((sys_reg->I2CCLK >> offset) & SYSTEM_I2CCLK_I2C0CLKSRC_Msk) ? PLL_FREQ
+										   : RC25M_FREQ;
+
+		offset = SYSTEM_I2CCLK_I2C0CLKDIV_Pos;
+		offset += ((SYSTEM_I2CCLK_I2C1CLKDIV_Pos - SYSTEM_I2CCLK_I2C0CLKDIV_Pos) *
+			   module_idx);
+		divide = ((sys_reg->I2CCLK >> offset) & SYSTEM_I2CCLK_I2C0CLKDIV_Msk);
+
+		freq = (src >> divide);
+
+		LOG_DBG("I2C%d: src<%s> divide<%ld> freq<%d>", module_idx,
+			(src == PLL_FREQ) ? "PLL" : "RC25M", (0x1ul << divide), freq);
+		break;
+	case RTS5912_SCCON_UART:
+		src = (sys_reg->UARTCLK & SYSTEM_UARTCLK_SRC_Msk) ? PLL_FREQ : RC25M_FREQ;
+		divide = ((sys_reg->UARTCLK & SYSTEM_UARTCLK_DIV_Msk) >> SYSTEM_UARTCLK_DIV_Pos);
+		freq = (src >> divide);
+
+		LOG_DBG("UART0: src<%s> divide<%ld> freq<%d>", (src == PLL_FREQ) ? "PLL" : "RC25M",
+			(0x1ul << divide), freq);
+		break;
+	case RTS5912_SCCON_ADC:
+		src = (sys_reg->ADCCLK & SYSTEM_ADCCLK_SRC_Msk) ? RC25M_FREQ : PLL_FREQ;
+		divide = ((sys_reg->ADCCLK & SYSTEM_ADCCLK_DIV_Msk) >> SYSTEM_ADCCLK_DIV_Pos);
+
+		switch (divide) {
+		case 0:
+		case 1:
+		case 2:
+		case 3:
+			freq = (src / (divide + 1));
+			divide += 1;
+			break;
+		case 4:
+			freq = (src / 6);
+			divide = 6;
+			break;
+		case 5:
+			freq = (src / 8);
+			divide = 8;
+			break;
+		case 6:
+			freq = (src / 12);
+			divide = 12;
+			break;
+		case 7:
+			freq = (src / 16);
+			divide = 16;
+			break;
+		default:
+			return -EINVAL;
+		}
+
+		LOG_DBG("ADC0: src<%s> divide<%d> freq<%d>", (src == PLL_FREQ) ?
+				"PLL" : "RC25M", divide, freq);
+		break;
+	case RTS5912_SCCON_SYS:
+		src = (sys_reg->SYSCLK & SYSTEM_SYSCLK_SRC_Msk) ? PLL_FREQ : RC25M_FREQ;
+		divide = (sys_reg->SYSCLK & SYSTEM_SYSCLK_DIV_Msk) ? 0x01ul : 0x00ul;
+		freq = (src >> divide);
+
+		LOG_DBG("System Clock: src<%s> divide<%d> freq<%d>",
+			(src == PLL_FREQ) ? "PLL" : "RC25M", divide, freq);
+		break;
+	case RTS5912_SCCON_PERIPH_GRP0:
+	case RTS5912_SCCON_PERIPH_GRP1:
+	case RTS5912_SCCON_PERIPH_GRP2:
+		LOG_ERR("Not support peripheral group #%d-%d", clk_grp, clk_idx);
+		return -ENOTSUP;
+	default:
+		LOG_ERR("Unknown peripheral group #%d", clk_grp);
+		return -EINVAL;
+	}
+
+	*rate = freq;
+	return 0;
+}
+
+static struct clock_control_driver_api rts5912_clock_control_api = {
+	.on = rts5912_clock_control_on,
+	.off = rts5912_clock_control_off,
+	.get_rate = rts5912_clock_control_get_rate,
+};
+
+static int rts5912_clock_control_init(const struct device *dev)
+{
+	const struct rts5912_sccon_config *const config = dev->config;
+	SYSTEM_Type *sys_reg = (SYSTEM_Type *)config->reg_base;
+
+	/* Setup system clock */
+	sys_reg->SYSCLK = ((config->sysclk_src << SYSTEM_SYSCLK_SRC_Pos) |
+			   (config->sysclk_div << SYSTEM_SYSCLK_DIV_Pos));
+	return 0;
+}
+
+const struct rts5912_sccon_config rts5912_sccon_config = {
+	.reg_base = DT_INST_REG_ADDR_BY_IDX(0, 0),
+	.sysclk_src = (uint8_t)DT_INST_PROP_OR(0, sysclk_src, 1),
+	.sysclk_div = (uint8_t)DT_INST_PROP_OR(0, sysclk_div, 0),
```

Comment:
`sysclk-src` and `sysclk-div` are not properties of `realtek,rts5912-sccon`.  This should be deleted or just set to hard-coded value if you plan to add these properties later.

Reply:
We will set this to hard-coded right now.